src/js/common/schemaform/reducers/index.js

```diff
@@ -147,35 +153,61 @@ export default function createSchemaFormReducer(formConfig) {
       }
       case SET_SAVE_FORM_STATUS: {
         const newState = _.set('savedStatus', action.status, state);
+        newState.startingOver = false;
+        newState.prefillStatus = PREFILL_STATUSES.notAttempted;
 
         // This is the only time we have a saved datetime
         if (action.status === SAVE_STATUSES.success) {
-          return _.set('lastSavedDate', action.lastSavedDate, newState);
+          newState.lastSavedDate = action.lastSavedDate;
         }
 
         return newState;
       }
       case SET_FETCH_FORM_STATUS: {
         return _.set('loadedStatus', action.status, state);
       }
+      case SET_FETCH_FORM_PENDING: {
+        const newState = _.set('loadedStatus', LOAD_STATUSES.pending, state);
+
+        if (action.prefill) {
+          newState.prefillStatus = PREFILL_STATUSES.pending;
+        }
+
+        return newState;
+      }
       case SET_IN_PROGRESS_FORM: {
         let newState;
 
         // if we're prefilling, we want to use whatever initial data the form has
-        if (action.prefilled) {
+        if (state.prefillStatus === PREFILL_STATUSES.pending) {
           const formData = _.merge(state.data, action.data.formData);
           const loadedData = _.set('formData', formData, action.data);
           newState = _.set('loadedData', loadedData, state);
+          newState.prefillStatus = PREFILL_STATUSES.success;
         } else {
           newState = _.set('loadedData', action.data, state);
+          newState.prefillStatus = PREFILL_STATUSES.notAttempted;
         }
 
         newState.loadedStatus = LOAD_STATUSES.success;
-        newState.prefilled = action.prefilled;
         newState.data = newState.loadedData.formData;
 
         return recalculateSchemaAndData(newState);
       }
+      case SET_START_OVER: {
+        return _.assign(state, {
+          isStartingOver: true,
+          data: state.initialData,
+          loadedStatus: LOAD_STATUSES.pending
+        });
+      }
+      case SET_PREFILL_COMPLETE: {
+        return _.assign(state, {
+          prefillStatus: PREFILL_STATUSES.complete,
```

Comment:
Yeah, it might be better to change that complete status to failed or something. It initially covered both success and failure.